clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -29,27 +30,50 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsNull.nullValue;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
 
 public class SerializationTest {
 
     final private String topic = "testTopic";
+    final private Map<Class<Object>, List<Object>> testData = new HashMap() {
+        {
+            put(String.class, Arrays.asList("my string"));
+            put(Short.class, Arrays.asList((short) 32767, (short) -32768));
+            put(Integer.class, Arrays.asList((int) 423412424, (int) -41243432));
+            put(Long.class, Arrays.asList(922337203685477580L, -922337203685477581L));
+            put(Float.class, Arrays.asList(5678567.12312f, -5678567.12341f));
+            put(Double.class, Arrays.asList(5678567.12312d, -5678567.12341d));
+            put(byte[].class, Arrays.asList("my string".getBytes()));
+            put(ByteBuffer.class, Arrays.asList(ByteBuffer.allocate(10).put("my string".getBytes())));
+            put(Bytes.class, Arrays.asList(new Bytes("my string".getBytes())));
+        }
+    };
 
     private class DummyClass {
-
     }
 
     @Test
-    public void testSerdeFrom() {
-        Serde<Long> thisSerde = Serdes.serdeFrom(Long.class);
-        Serde<Long> otherSerde = Serdes.Long();
-
-        Long value = 423412424L;
+    public void testAllSerdes() {
+        for (Map.Entry<Class<Object>, List<Object>> test : testData.entrySet()) {
+            try (Serde<Object> serde = Serdes.serdeFrom(test.getKey())) {
+                for (Object value : test.getValue()) {
+                    assertEquals("Should get the original " + test.getKey().getSimpleName() +
+                                    " after serialization and deserialization", value,
+                            serde.deserializer().deserialize(topic, serde.serializer().serialize(topic, value)));
+                }
+            }
+        }
+    }
 
-        assertEquals("Should get the original long after serialization and deserialization",
-                value, thisSerde.deserializer().deserialize(topic, otherSerde.serializer().serialize(topic, value)));
-        assertEquals("Should get the original long after serialization and deserialization",
-                value, otherSerde.deserializer().deserialize(topic, thisSerde.serializer().serialize(topic, value)));
+    @Test
+    public void serdeShouldSupportNull() {
```

Comment:
Perhaps: `allSerdesShouldSupportNull()`

Btw, I like the addition of these type of tests, which are essentially law checks.